sdk/api_client.go

```diff
@@ -54,9 +58,262 @@ func NewClientWithClienter(host string, httpClient HTTPClient, opts Options) *AP
 	}
 }
 
+// == Roles Endpoint ==
+
+func (c *APIClient) GetRoles(ctx context.Context) (*models.Roles, error) {
+	uri := fmt.Sprintf(rolesEndpoint, c.host)
+
+	log.Info(ctx, "GetRoles: starting permissions get policy request", log.Data{"uri": uri})
+
+	req, err := http.NewRequest(http.MethodGet, uri, nil)
```

Comment:
We should be setting request headers to include authorisation header; the Options struct can be extended to include an AuthorizationHeader key of type string

```
type header string

const (
	// List of available headers
	Authorisation header = "Authorisation"
)

// Options is a struct containing for customised options for the API client
type Options struct {
	Headers map[header][]string
}
```

Additional code logic:
```
func <method signature> {
...
        req, err := http.NewRequest(http.MethodGet, uri, nil)
	if err != nil {
		log.Info(ctx, "GetPermissionsBundle: error building new request", log.Data{"err": err.Error()})
		return nil, err
	}

	if len(c.options.Headers) > 0 {
		setHeaders(req, c.options.Headers)
	}
...
}

func setHeaders(req *http.Request, headers map[header][]string) {
	for h := range headers {
		for i := range h {
			req.Header.Add(string(h), headers[h][i])
		}
	}
}
```